### PR TITLE
fix: set correct nonce in EthCall/EstimateGas

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #870

## Description

for contract creation tx, the executing msg must contain the correct nonce value, because `ApplyMessage` will reset sender's nonce to that first.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (evm) [tharsis#838](https://github.com/tharsis/ethermint/pull/838) Fix splitting of trace.Memory into 32 chunks.
 * (rpc) [tharsis#860](https://github.com/tharsis/ethermint/pull/860) Fix `eth_getLogs` when specify blockHash without address/topics, and limit the response size.
 * (rpc) [tharsis#865](https://github.com/tharsis/ethermint/pull/865) Fix RPC Filter parameters being ignored
+* (evm) [tharsis#]() Set correct nonce in `EthCall` and `EstimateGas` grpc query.
```
Comment (fedekunze) [inline CHANGELOG.md:69]: Missing link 👍

**Review by fedekunze — APPROVED** [review]
utACK, can you add a test to ensure this doesn't break again? 🙏

File: CHANGELOG.md (hunk 1 above)
Comment (yihuang) [inline CHANGELOG.md:69]:
```suggestion
* (evm) [tharsis#871](https://github.com/tharsis/ethermint/pull/871) Set correct nonce in `EthCall` and `EstimateGas` grpc query.
```

Comment (yihuang) [reply CHANGELOG.md]: done

**Comment by yihuang** [comment]
> utACK, can you add a test to ensure this doesn't break again? 🙏

unit test added, I checked that the test fail on main branch.

**Pull request merged**